rio/transmat/impl/git2/git_transmat.go

```diff
@@ -0,0 +1,205 @@
+
+type GitTransmat struct {
+	workArea workArea
+}
+
+var _ rio.TransmatFactory = New
+
+func New(workPath string) rio.Transmat {
+	mustDir(workPath)
+	workPath, err := filepath.Abs(workPath)
+	if err != nil {
+		panic(meep.Meep(
+			&rio.ErrInternal{Msg: "Unable to set up workspace"},
+			meep.Cause(err),
+		))
+	}
+	wa := workArea{
+		fullCheckouts:  filepath.Join(workPath, "full"),
+		gitStorageDirs: filepath.Join(workPath, "gits"),
+	}
+	mustDir(wa.fullCheckouts)
+	mustDir(wa.gitStorageDirs)
+	return &GitTransmat{wa}
+}
+
+func selectWarehouse(log log15.Logger, siloURIs []rio.SiloURI) *Warehouse {
+	// Our policy is to take the first path that exists.
+	//  This lets you specify a series of potential locations,
+	//  and if one is unavailable we'll just take the next.
+	// Future work: cycle through later potential locations if one returns DNE!
+	//  (Unfortunately this is tricky to implement efficiently with git commands.)
+	if len(siloURIs) < 1 {
+		panic(&def.ErrWarehouseUnavailable{
+			Msg:    "No warehouse coords configured!",
+			During: "fetch",
+		})
+	}
+	var warehouse *Warehouse
+	for _, uri := range siloURIs {
+		wh := NewWarehouse(uri)
+		pong := wh.Ping()
+		if pong == nil {
+			log.Info("git: connected to remote warehouse", "remote", uri)
+			warehouse = wh
+			break
+		} else {
+			log.Info("Warehouse unavailable, skipping",
+				"remote", uri,
+				"reason", pong,
+			)
+		}
+	}
+	if warehouse == nil {
+		panic(&def.ErrWarehouseUnavailable{
+			Msg:    "No warehouses responded!",
+			During: "fetch",
+		})
+	}
+	return warehouse
+}
+
+/*
+	Git transmats plonk down the contents of one commit (or tree) as a filesystem.
+
+	A fileset materialized by git does *not* include the `.git` dir by default,
+	since those files are not themselves part of what's described by the hash.
+
+	Git effectively "filters" out several attributes -- permissions are only loosely
+	respected (execution only), file timestamps are undefined, uid/gid bits
+	are not tracked, xattrs are not tracked, etc.  If you desired defined values,
+	*you must still configure materialization to use a filter* (particularly for
+	file timestamps, since they will otherwise be allowed to vary from one
+	materialization to the next(!)).
+
+	Git also allows for several other potential pitfalls with lossless data
+	transmission: git cannot transmit empty directories.  This can be a major pain.
+	Typical workarounds include creating a ".gitkeep" file in the empty directory.
+	Gitignore files may also inadventantly cause trouble.  Transmat.Materialize
+	will act *consistently*, but it does not overcome these issues in git
+	(doing so would require additional metadata or protocol extensions).
+
+	This transmat is *not* currently well optimized, and should generally be assumed
+	to be re-cloning on all materializations -- specifically, it is not smart
+	enough to recognize requests for different commits and trees from the
+	same repos in order to save reclones.
```

Comment:
Is this comment still true?  (I think it's copypasta from the old git transmat, for which it is also now out of date!  I'll fix that in another branch...)